Sources/Testing/Running/Runner.Plan.swift

```diff
@@ -159,17 +159,27 @@ extension Runner.Plan {
     // them, in which case it will be .recordIssue().
     var testGraph = Graph<String, Test?>()
     var actionGraph = Graph<String, Action>(value: .run)
-    for test in tests where _isTestIncluded(test, using: configuration.testFilter) {
+    for test in tests {
       let idComponents = test.id.keyPathRepresentation
       testGraph.insertValue(test, at: idComponents)
       actionGraph.insertValue(.run, at: idComponents, intermediateValue: .run)
```

Comment:
Moving the `where ...` clause here will cause the `actionGraph` to always be updated, even if some of those tests are later filtered out. Does something else later on cause the corresponding nodes in that graph to be pruned?

Reply:
Yes, immediately below we prune them out (after applying traits recursively) on line 173.